{% extends "base.html" %}
{% load static i18n %}


{% block content %}
    <div class="container">

        <div class="row my-5">
            <div class="col-md-6 mb-5">
                <h2>Fork us</h2>
                <hr>
                <p>	The web interface of the ReproGenomics Viewer database was built with two open web frameworks, <a href="https://trypyramid.com/" target="_blanck">Pyramid</a> and <a href="https://angularjs.org/" target="_blanck">AngularJS</a>.
                    Pyramid embeds many features, such as a REST API, a JSON renderer, an SQLAlchemy Object-relational mapper (ORM), a Deform library to generate forms, and compatibility with SMTP servers.
                    AngularJS, on the other hand, is an open JavaScript framework that extends traditional HTML vocabulary; it allows implementation of readable and quickly developable web environments.</p>
                    The full source code of the ReproGenomics Viewer web application is available <a href="https://github.com/fchalmel/RGV">here</a>.
            
            </div>

            <div class="col-md-6 mb-5">
                <h2>Data visualization</h2>
                <hr>
                <p>All charts generated by RGV's tools use <a href="https://plot.ly/javascript/">plotly.js plugin</a>. Built on top of <a href="https://d3js.org/">d3.js</a> and <a href="http://stack.gl/">stack.gl</a>, 
                    plotly.js is a high-level, declarative charting library. plotly.js ships with 20 chart types, including 3D charts, statistical graphs, and SVG maps. </p>
            
            </div>
        </div>

        <div class="row my-5">

            <div class="col-md-6 mb-5">
                <h2>RGV Backbone</h2>
                <hr>
                <p>The backbone of the RGV is the series of tools for processing and organizing data within the system.
                    The system integrates an implementation of the <a href="https://www.ncbi.nlm.nih.gov/pubmed/27072794" target="_blanck">JBrowse</a> genome browser grafted to the RGV backbone.<br></p>
                    <p>Four types of information are manually extracted and curated for each study, including:</p>
                    <ul class="list-unstyled">
                        <li>1- the scientific name of each species and the genome release with which the experiments were performed and analyzed;</li>
                        <li>2- the associated scientific publication;</li>
                        <li>3- each biology topic investigated in the study;</li>
                        <li>4- and the high-throughput technologies performed. Then each raw data file of a given data set underwent a series of automatic processing to make it fully compatible with the RGV system.</li>
                    </ul>
            
            </div>
        
            <div class="col-md-6 mb-5">
                <h2>Sample Procesing</h2>
                <hr>
                <p>For each sample, eight processing steps are sequentially performed:</p>
                <ul class="list-unstyled">
                    <li>1- The raw data file (sra file) of a given sample and their associated information are downloaded from the <a href="https://www.ncbi.nlm.nih.gov/geo/" target="_blanck">Gene Expression Omnibus</a> (GEO) and to the <a href="https://www.ncbi.nlm.nih.gov/sra" target="_blanck">Sequence Read Archive</a> (SRA);</li>
                    <li>2- The fastq file is extracted from the sra archive file;</li>
                    <li>3- reads are mapped on its corresponding reference genome using the <a href="https://www.ncbi.nlm.nih.gov/pubmed/23104886" target="_blanck">STAR</a> tool.</li>
                    <li>4- each bam file is converted into a simple tab-delimited text file (BED) in which data are standardized according to the total nuber of read counts using <a href="https://www.ncbi.nlm.nih.gov/pubmed/20110278" target="_blanck">BEDTools</a>;</li>
                    <li>5- the standardized BED file is then converted into an indexed binary format (bigWig or bw) to enable fast remote access to the data;</li>
                    <li>6- the pairwise alignments between genome assemblies and between species provided by UCSC made it possible to convert genome coordinates in the resulting bigWig file from a genome of a given species into a genome of another species ussing <a href="https://www.ncbi.nlm.nih.gov/pubmed/24351709" target="_blanck">CrossMap</a>.</li>
                    <li>7- reference transcripts are quantified in the corresponding sample using the <a href="https://www.ncbi.nlm.nih.gov/pubmed/25690850" target="_blanck">StringTie</a> tool.</li>
                    <li>8- We used manually extracted information to organize the processed data into four broad categories, i.e. biological topics, technologies, publications and species.</li>
                </ul>

            </div>
        </div>
        
    </div>

{% endblock %}
